opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -230,7 +230,7 @@ private Optional<ExprType> type(String field) {
   private static ExprValue parseDateTimeString(String value, OpenSearchDateType dataType) {
     List<DateFormatter> formatters = dataType.getAllNamedFormatters();
     formatters.addAll(dataType.getAllCustomFormatters());
-    ExprCoreType returnFormat = (ExprCoreType) dataType.getExprType();
+    ExprCoreType returnFormat = dataType.getExprCoreType();
```

Comment:
The `returnFormat` could be `UNKNOWN` now. Be mind that `getExprCoreType()` may introduce some behaviour changes. Maybe it's safe here.

Reply:
I think with the current code too returnFormat could have been unknown. Here I didn't see any issues with changing to `ExprCoreType`.